packages/connect/src/api/applySettings.ts

```diff
@@ -34,7 +34,33 @@ export default class ApplySettings extends AbstractMethod<'applySettings', PROTO
 
     async run() {
         const cmd = this.device.getCommands();
-        const response = await cmd.typedCall('ApplySettings', 'Success', this.params);
+
+        // https://github.com/trezor/trezor-firmware/pull/5015
+        // homescreen bytes are streamed in smaller data chunks
+        const homescreenBytes = this.params.homescreen
+            ? Buffer.from(this.params.homescreen, 'hex')
+            : undefined;
+        if (this.device.atLeast('2.9.0') && homescreenBytes) {
```

Comment:
so, 2.9.0 already supports this feature? we might want to cherry-pick this to the current release? @Hannsek

Reply:
this is not breaking change, the the old way (1 message with `homescreen` as bytes)  is working as it used to.
this is relevant only for THP builds where message size was reduced to ~6kB